protocol/message.go

```diff
@@ -52,48 +52,77 @@ type MonitoringRequest struct {
 }
 
 // Response messages
-type Response interface{}
-
-type ErrorResponse struct {
-	Error ErrorCode
+type Response struct {
```

Comment:
I thought about it but in case the user requested an invalid message type, which's type the server should return in the response?